### DISCUSSION TITLE
SQLite is missed

### DISCUSSION THREAD

**Discussion by b-a-p** [Brewtarget/brewtarget#637] [Q&A] [answered]
Hi,
I just installed latest prebuilt on Windows 10 but when I run it - it throws an error: "could not open SQLite DB file. Driver not loaded". Where can I find instruction how overcome it?

**Comment by matty0ung** [comment]
Is this the same issue as https://github.com/Brewtarget/brewtarget/issues/632?

**Reply by mattiasmaahl** [reply]
Yes I would say that it is.
I can share the Msi installer again for testing?

**Comment by mattiasmaahl** [comment]
I will share the MSI installer again tomorrow an push a PR for fixing the build action on github. I suspct that this is a QT version Issue.

**Comment by mattiasmaahl** [comment]
@b-a-p You will find the latest version of Brewtarget develop on the MSI below. that one is built with QT5.15.1 instead of the latest version 5.15.2.
My conclusion from issue #632 was that the there is something wonky about the latest version in QT.
I was waiting for @MTkeg to revert if the fix was working for him so I could push a PR to fix the build actions.
But please test this and revert if it is working for you.

[Download Brewtarget 2.4.0 QT 5.15.1](https://owncloud.maahl.com/index.php/s/sOqsy48vk7ui0Mn)

**Reply by b-a-p** [reply]
Hi @mattiasmaahl,
Many thanks - version provided by you is working fine! But it has interface which is too big - it is not fit on full HD monitor, I compared it to version 2.3  from http://www.brewtarget.org/download.html

**Reply by mattiasmaahl** [reply]
Good to hear, could you supply screenshots on the GUI you're experiencing?

**Reply by b-a-p** [reply]
Hi @mattiasmaahl,
many thanks for your support, after restart everything functioning so far.

**Comment by b-a-p** [comment]
Hi @mattiasmaahl,
Unfortunately version provided by you has some issues, for example with mash creation. Can you please give us estimation when you expect to solve the issue https://github.com/Brewtarget/brewtarget/issues/632? And also I'm wondering why there is no released 2.4 version?
P.S.:
no pusing, just asking :)

**Reply by matty0ung** [reply]
There are a couple of fixes waiting to be merged which may address the mash creation issues.  Will chat with @mattiasmaahl about getting these code reviewed.

Release of 2.4 is partly "We need to get around to it" and partly "We need to find out how to".  But I agree we should try to get to it sooner rather than later, as it's been a while since 2.3.

**Reply by matty0ung** [reply]
FYI, I have merged into dev the fix for the mash creation issues.

**Reply by b-a-p** [reply]
Hi @mattiasmaahl,
Many thanks for the update. But what about QT/SQLite bug - do you have any idea when it can be fixed?

**Reply by mattiasmaahl** [reply]
It is still in progress. I have been busy as of late as I changed jobs. Will get around to it soon.

**Comment by b-a-p** [comment]
Hi @mattiasmaahl,
I'm sorry for nagging but I downloaded latest build and the error is still there ;(

**Reply by matty0ung** [reply]
I am currently trying to get the Windows build to do what I want, so am hopeful we will have some progress to report in the not too distant future.

Most of the time, the magic of Qt and CMake means we can write one bit of code that works on Windows, Mac and Linux.  But when it comes to installing and packaging, we have to get down into more of the platform-specific details.  I've now got a build running in a Windows VM on my Linux machine, so it's a bit quicker to test things than before (when I was relying on GitHub's Windows VMs).

**Reply by matty0ung** [reply]
Making progress.  I have it working on my machine.  Hope to post something this week.

**Reply by b-a-p** [reply]
Many thanks for the update,🤞.

**Comment by matty0ung** [comment]
I feel like I am making progress.  After doing some ugly CMake things, I have a build running on Windows.

When I double-click on the executable, I get the "driver not loaded" error.  However, when I run the same executable from the MSYS2/MinGW command line, the program starts fine.  That makes me wonder if it's something to do with PATH or copying a file to the right location.

Investigation continues...

**Reply by MTkeg** [reply]
Thank you all for looking into and working through the issues!  Sorry @mattiasmaahl and @matty0ung for dropping the ball on fix.  Work has been very hectic and doesn't appear to be getting any better; I haven't even brewed any beer or wine this year.  I hope the issue gets worked out; I like this software better then any of the others.

MT

**Comment by matty0ung** [comment, marked as answer]
Hi folks,
I've just pushed version 3.0.0 live.  I'm hoping it fixes this problem (based on the fact that it seems to work on my machine!).  As ever, please don't hesitate to let us know if it doesn't or if you hit any other problems.

**Reply by b-a-p** [reply]
Many thanks @matty0ung, now it runs on Windows 11!

**Comment by matty0ung** [comment]
Archiving this discussion, since the underlying problem was resolved.
